flamingpy/codes/graphs/egraph.py

```diff
@@ -214,3 +217,149 @@ def draw_adj(self, **kwargs):
 
         adj = self.adj_generator(sparse=False)
         return plot_mat_heat_map(adj, **kwargs)
+
+    def add_node(self, node: tuple, **kwargs):
+        """Overwrite add_node function from nx.Graph to raise a warning
+        whenever add_qubit should be used."""
+        if not (self.to_indices is None and self.adj_mat is None):
+            warnings.warn(
+                "EGraph attributes are already populated. To add a"
+                " node, please use the add_qubit function."
+            )
+        super().add_node(node, **kwargs)
+
+    def remove_node(self, node: tuple):
+        """Overwrite remove_node function from nx.Graph to raise a warning
+        whenever remove_node should be used."""
+        if not (self.to_indices is None and self.adj_mat is None):
+            warnings.warn(
+                "EGraph attributes are already populated. To remove a"
+                " node, please use the remove_qubit function."
+            )
+        super().remove_node(node)
+
```

Comment:
Come to think of it, we would also need to overwrite the functions add_nodes_from, remove_nodes_from, add_edges_from, remove_edges_from. Maybe we should just remove these to avoid the hassle? What do you think?

Reply:
Sounds good, I agree! I will remove these functions